### ISSUE TITLE
Build Error in VS 2017 - fatal error C1083: Cannot open include file: 'openssl/ssl.h': No such file or directory

### ISSUE THREAD

**Issue by FredComploier** [obgm/libcoap#827]
## Environment

Visual Studio 2017

## Problem Description

I found this in the Build Document
You may need to edit win32\libcoap.props to update the OpenSSLRootDir and
OpenSSLRootDirDbg variables to point to the top level directory where OpenSSL
is installed so that the include, lib etc. directories are correctly set up.

I searched for ssl.h file and it is found here:
C:\Program Files\OpenSSL-Win64\include\openssl

I understand what it needs, now how do I modify the libcoap.props below to point correctly?

     <OpenSSLRootDir>$(OPENSSL_INSTALL_PATH)</OpenSSLRootDir>
    <OpenSSLRootDirDbg>$(OPENSSL_INSTALL_PATH)</OpenSSLRootDirDbg>
    <OpenSSLIncludeDir>$(OpenSSLRootDir)include</OpenSSLIncludeDir>
    <OpenSSLIncludeDirDbg>$(OpenSSLRootDirDbg)include</OpenSSLIncludeDirDbg>
    <OpenSSLLibDir>$(OpenSSLRootDir)lib</OpenSSLLibDir>
    <OpenSSLLibDirDbg>$(OpenSSLRootDirDbg)lib</OpenSSLLibDirDbg>

**Comment by mrdeep1** [comment]
I would try settimng `OPENSSL_INSTALL_PATH `as a windows environment variable with the value `"C:\Program Files\OpenSSL-Win64\"` and then restart VS 2017.

Alternatively, you can make a local modification replacing the 2 instances of `$(OPENSSL_INSTALL_PATH)` with `C:\Program Files\OpenSSL-Win64\`    - note the trailing \ in both suggestions.

**Comment by mrdeep1** [comment]
I'm surprised that Option 1 did not work, it does on my Windows device
````
C:\Users\mrdeep1>set
...
OPENSSL_INSTALL_PATH=C:\Program Files\OpenSSL-Win64\
...
````
which means I do not need to edit `libcoap.props`.

I currently do not have OPenssl 3.0 on my windows device, so am not able to look at the SHA256_Init issue at present. I am working on a fix for the other reported errors which were introduced in a patch and not properly checked out is a VS environment.

No need to raise a separate issue.

**Comment by mrdeep1** [comment]
PLease checkout #829 which should also fix your OpenSSL 3.0 issue.  Please let us know how it goes.
